modAionImpl/src/org/aion/zero/impl/db/AionRepositoryImpl.java

```diff
@@ -866,6 +868,42 @@ public void compactState() {
         }
     }
 
+    /**
+     * Imports a trie node to the indicated blockchain database.
+     *
+     * @param key the hash key of the trie node to be imported
+     * @param value the value of the trie node to be imported
+     * @param dbType the database where the key-value pair should be stored
+     * @throws IllegalArgumentException if the given key is null or the database type is not
+     *     supported
+     * @return a {@link TrieNodeResult} indicating the success or failure of the import operation
+     */
+    public TrieNodeResult importTrieNode(byte[] key, byte[] value, DatabaseType dbType) {
+        IByteArrayKeyValueDatabase db = selectDatabase(dbType);
+
+        Optional<byte[]> stored = db.get(key);
+        if (stored.isPresent()) {
+            if (Arrays.equals(stored.get(), value)) {
+                return TrieNodeResult.KNOWN;
+            } else {
+                return TrieNodeResult.INCONSISTENT;
+            }
+        }
+
+        // empty keys are not allowed
+        if (key == null || key.length != V1Constants.HASH_SIZE) {
+            return TrieNodeResult.INVALID_KEY;
+        }
+
+        // not allowing deletions to be imported
+        if (value == null || value.length == 0) {
+            return TrieNodeResult.INVALID_VALUE;
+        }
```

Comment:
why not check the key value in the beginning?

Reply:
good point, I'll update it